src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +55,190 @@ class MissingProjectMembership extends React.Component<Props, State> {
+      }
+      return <StyledButton busy>Request Access</StyledButton>;
+    } else if (teamStatus?.isPending) {
+      return <StyledButton disabled>Request Pending</StyledButton>;
     } else if (features.has('open-membership')) {
       return (
-        <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
-          {t('Join Team')}
-        </a>
+        <StyledButton priority="primary" onClick={this.joinTeam.bind(this, team)}>
+          Join Team
+        </StyledButton>
       );
     }
     return (
-      <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
-        {t('Request Access')}
-      </a>
+      <StyledButton priority="primary" onClick={this.joinTeam.bind(this, team)}>
+        Request Access
+      </StyledButton>
     );
   }
 
-  renderExplanation(features: Set<string>) {
-    if (features.has('open-membership')) {
-      return t('To view this data you must one of the following teams.');
-    } else {
-      return t(
-        'To view this data you must first request access to one of the following teams:'
-      );
-    }
-  }
-
-  renderJoinTeams(features: Set<string>) {
+  requestAccess() {
+    const request: string[] = [];
+    const pending: string[] = [];
     const teams = this.state.project?.teams ?? [];
-
-    if (!teams.length) {
-      return (
-        <EmptyMessage>
-          {t(
-            'No teams have access to this project yet. Ask an admin to add your team to this project.'
-          )}
-        </EmptyMessage>
-      );
-    }
-
-    return teams.map((team: Team) => (
-      <p key={team.slug}>
-        #{team.slug}: {this.renderJoinTeam(team, features)}
-      </p>
-    ));
+    const team = teams.map(tm => TeamStore.getBySlug(tm.slug));
+    team.map(tm => (tm?.isPending ? pending.push(tm!.slug) : request.push(tm!.slug)));
+    return [request, pending];
   }
 
+  handleChange = evt => {
+    const input = evt;
+    this.setState({team: input});
+  };
+
+  requestPendingTeam = team => {
+    return {
+      value: team,
+      label: (
+        <StyledTooltip position="left" title={t(`Request pending for #${team}`)}>
+          <UnmentionableTeam>
+            <DisabledLabel>{`#${team}`}</DisabledLabel>
+          </UnmentionableTeam>
+        </StyledTooltip>
+      ),
+    };
+  };
+
   render() {
-    const {organization} = this.props;
+    const {organization, groupId} = this.props;
+    const team = this.state.team;
+    const teams = this.state.project?.teams ?? [];
     const features = new Set(organization.features);
 
+    const endpoint = `/organizations/${organization.slug}/members/me/`;
+
+    const teamAccess = [
+      {
+        label: 'Request Access',
+        options: this.requestAccess()[0].map(request => ({
+          value: request,
+          label: `#${request}`,
+        })),
+      },
+      {
+        label: 'Pending Requests',
+        options: this.requestAccess()[1].map(pending => this.requestPendingTeam(pending)),
```

Comment:
I would pick a different name here for `requestPendingTeam` since it sounds like we are making a request for the pending team, but we are just getting the `option` for a pending team.

Reply:
I renamed this to `renderPendingTeam`

Reply:
nit: we use `render` to mean that it's returning a react element, this is returning a object that's used for the Select options